src/TestEnvironment.Docker.Containers.MariaDB/MariaDBContainerWaiter.cs

```diff
@@ -1,44 +1,39 @@
 ﻿using System;
 using System.Threading;
 using System.Threading.Tasks;
-using TestEnvironment.Docker;
 using Microsoft.Extensions.Logging;
 using MySql.Data.MySqlClient;
 
 namespace TestEnvironment.Docker.Containers.MariaDB
 {
-    public class MariaDBContainerWaiter : IContainerWaiter<MariaDBContainer>
+    public class MariaDBContainerWaiter : BaseContainerWaiter<MariaDBContainer>
     {
-        private readonly ILogger _logger;
-
         public MariaDBContainerWaiter(ILogger logger = null)
+            : base(logger)
         {
-            _logger = logger;
         }
 
-        public async Task<bool> Wait(MariaDBContainer container, CancellationToken cancellationToken)
+        protected override async Task<bool> PerformCheckAsync(MariaDBContainer container, CancellationToken cancellationToken)
         {
-            if (container == null) new ArgumentNullException(nameof(container));
-
             try
             {
-                using (var connection = new MySqlConnection(container.GetConnectionString()))
-                using (var command = new MySqlCommand("select @@version", connection))
-                {
-                    await command.Connection.OpenAsync();
-                    await command.ExecuteNonQueryAsync();
-                }
+                // ReSharper disable UseAwaitUsing
```

Comment:
in case of using `await using` I got an exception `Method not found: 'System.Threading.Tasks.Task MySql.Data.MySqlClient.MySqlConnection.DisposeAsync()'.` on .Net Core 3.1. I'm curious why it's okay for .Net Core 2.1, but decided to not use `await using` in such case.

Reply:
In such case it would be better to revert changes here and leave old using block style

Reply:
remove ReSharper related comments